pkg/storage/series/series_set.go

```diff
@@ -89,21 +90,40 @@ func NewConcreteSeriesIterator(series *ConcreteSeries) chunkenc.Iterator {
 	}
 }
 
-func (c *concreteSeriesIterator) Seek(t int64) bool {
+func (c *concreteSeriesIterator) Seek(t int64) chunkenc.ValueType {
 	c.cur = sort.Search(len(c.series.samples), func(n int) bool {
 		return c.series.samples[n].Timestamp >= model.Time(t)
 	})
-	return c.cur < len(c.series.samples)
+	if c.cur < len(c.series.samples) {
+		return chunkenc.ValFloat
+	}
+	return chunkenc.ValHistogram
```

Comment:
`Seek()` seems to be taken directly from Prometheus, and when it was modified to support the new iterator interface some more logic was added: https://github.com/prometheus/prometheus/pull/9877/files#diff-ee99cb62c74579879029165b587c656d03fd78290a3abb23b2e1d68fccd8c0b7R361. Should that also be added here?

Reply:
Both should technically be equvalent. I would leave out the re-work from here for now.
Also, should it not be ValNone here?

Reply:
Whoops, yeah.